### ISSUE TITLE
Do not generate empty body for implicitly external function declarations

### ISSUE THREAD

**Issue by NiematojakTomasz** [square/kotlinpoet#1021]
Sample of invalid generated code:
```
public actual abstract external class DOMPoint : DOMPointReadOnly {
  // ...
  public actual companion object {
    public actual fun fromPoint(other: DOMPointInit?): DOMPoint {
    }
  }
}
```
Compiling above code gives error:
Wrong body of external declaration. Must be either ' = definedExternally' or { definedExternally }

**Comment by JakeWharton** [comment]
Well your request is not just to not generate a body, it's also to generate `definedExternally`, right?

And I don't think that we should do this. I think you need to add `addStatement("return definedExternally")` here. We don't know whether you're generating for Kotlin/JS or simply made a mistake adding `external` to the enclosing class declaration. `definedExternally` is only available when targeting Kotlin/JS, so the compilation error you would get in a multiplatform, JVM, native, or WASM target about not being able to resolve `definedExternally` would likely hide the real problem. Whereas if you specify it yourself then it becomes clear of the origin of that symbol.

**Comment by NiematojakTomasz** [comment]
Compiler also accepts no body external declarations. I think there should be an option to generate such. For me explicitly returning `definedExternally` is just redundant for external declaration. Note - we don't mark properties with `definedExternally`.

**Comment by Egorand** [comment]
We're special-casing `EXTERNAL` `FunSpec`s already:

https://github.com/square/kotlinpoet/blob/f028a95776d0b58eda5b2492d157b8d39c88a939/kotlinpoet/src/main/java/com/squareup/kotlinpoet/FunSpec.kt#L101-L106

so I'd expect us to not emit the body. Is it a bug with the special-casing logic?

**Comment by JakeWharton** [comment]
I would be comfortable omitting the body of functions in external classes. I wouldn't want to automatically assume Kotlin/JS and use definedExternally, however.

**Comment by Egorand** [comment]
Yep, agreed. @NiematojakTomasz is there any difference between `fun foo()` and `fun foo() = definedExternally`, or they're equivalent?

**Comment by NiematojakTomasz** [comment]
@Egorand AFAIK there's no difference. But the only reference I could find is in [definedExternally](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.js/defined-externally.html) kdoc:

> To represent body of an external function. In most cases Kotlin does not require to provide bodies of external functions and properties, but if for some reason you want to (for example, due to limitation of your coding style guides), you should use definedExternally.

**Comment by Egorand** [comment]
I think we should not emit function/property bodies in `external` declarations, same as what we do for `expect`.

**Comment by Egorand** [comment]
Closed by #1026
